frontend/src/pages/modelServing/screens/global/DeleteInferenceServiceModal.tsx

```diff
@@ -29,16 +33,24 @@ const DeleteInferenceServiceModal: React.FC<DeleteInferenceServiceModalProps> =
   return (
     <DeleteModal
       title="Delete deployed model?"
-      isOpen={!!inferenceService}
+      isOpen={isOpen}
       onClose={() => onBeforeClose(false)}
       submitButtonLabel="Delete deployed model"
       onDelete={() => {
         if (inferenceService) {
           setIsDeleting(true);
-          deleteInferenceService(
-            inferenceService.metadata.name,
-            inferenceService.metadata.namespace,
-          )
+          Promise.all([
+            deleteInferenceService(
+              inferenceService.metadata.name,
+              inferenceService.metadata.namespace,
+            ),
+            servingRuntime
+              ? deleteServingRuntime(
```

Comment:
Be aware that there are several resources to delete with the serving runtime too, such as the rolebinding, secrets and more, I think we could cover this in https://github.com/opendatahub-io/odh-dashboard/issues/1998 but this logic should be similar to https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/pages/modelServing/screens/projects/DeleteServingRuntimeModal.tsx#L50

Reply:
The complex deletion of the ServingRuntime should be handled in #1998.

That being said, I am deleting two related resources here and I wonder if we should rely on creating an ownerReference between InferenceService and ServingRuntime for kserve. If so then I don't need to delete both resources here.